src/world/Network/Handlers/GMCommandHandlers.cpp

```diff
@@ -610,12 +622,26 @@ void Sapphire::Network::GameConnection::gm2Handler( FrameworkPtr pFw,
     }
     case GmCommand::Jump:
     {
+      player.prepareZoning( targetPlayer->getZoneId(), true, 1, 0 );
+      if( player.getCurrentInstance() )
+      {
+        player.exitInstance();
+      }
+      if( targetPlayer->getCurrentInstance() && player.getCurrentInstance() != targetPlayer->getCurrentInstance() )
+      {
+        auto instance = targetPlayer->getCurrentInstance();
+        auto pInstanceContent = instance->getAsInstanceContent();
+        // Not sure if GMs actually get bound to an instance they jump to on retail. It's mostly here to avoid a crash for now.
+        pInstanceContent->bindPlayer( player.getId() );
+        player.setInstance( targetPlayer->getCurrentInstance() );
+      }
       if( targetPlayer->getZoneId() != player.getZoneId() )
```

Comment:
```suggestion
      if( targetPlayer.getCurrentZone()->getGuid() != player.getCurrentZone()->getGuid() )
```
guess something like this?

Reply:
Yeah, I actually completely forgot about housing zones and it was indeed causing various issues and crashes left and right.

I changed it to what you suggested and it seems to be working fine now, I think I tested all cases.

There is one last thing bothering me (but I'm being a bit nitpicky here), `prepareZoning` doesn't have the time to fully fade out when moving to a different zone. It works fine when moving within the same zone and has a 1 second fade out, but it gets interrupted if moving to a different zone/instance. Is there a way to delay `setInstance` by 1 second or to have it ran only when `prepareZoning` is done?

Reply:
yeah there's queuedZoning which does that iirc, but I don't think it'll work with any zone you throw at it - I believe it just takes a territory type id

Reply:
https://github.com/SapphireServer/Sapphire/blob/aea0bf333a3b0fb3282d4ac3d24ad411b1613045/src/world/Actor/Player.h#L18-L33

So that'd work fine, but would need to be changed to take a zoneptr. I think we'll need to go through and change most stuff to queued zoning anyway because as is, a lot of stuff will instantly move you and it ruins muh immersion